Mathlib/GroupTheory/GroupAction/Blocks.lean

```diff
@@ -21,12 +24,46 @@ Given `SMul G X`, an action of a type `G` on a type `X`, we define
 
 The non-existence of nontrivial blocks is the definition of primitive actions.
 
+## Results for actions on finite sets
+
+- `IsBlock.ncard_block_mul_ncard_orbit_eq` : The cardinality of a block
+multiplied by the number of its translates is the cardinal of the ambient type
+
+- `IsBlock.is_top_of_large_block` : a too large block is equal to top
+
+- `IsBlock.is_subsingleton` : a too small block is a subsingleton
+
+- `IsBlock.of_subset` : the intersections of the translates of a finite subset
+that contain a given point is a block
+
 ## References
 
-We follow [wieland1964].
+We follow [wielandt1964].
 
 -/
 
+-- #find_home! doesn't find an adequate place for this
+theorem Setoid.nat_sum {α : Type _} [Finite α] {c : Set (Set α)} (hc : Setoid.IsPartition c) :
+    (finsum fun x : c => Set.ncard (x : Set α)) = Nat.card α := by
+  classical
+  have := Fintype.ofFinite α
+  simp only [finsum_eq_sum_of_fintype, Nat.card_eq_fintype_card, ← Set.Nat.card_coe_set_eq]
+  rw [← Fintype.card_sigma]
+  refine' Fintype.card_congr (Equiv.ofBijective (fun x => x.snd : (Σ a : ↥c, a) → α) _)
+  constructor
+  · -- injectivity
+    rintro ⟨⟨x, hx⟩, ⟨a, ha : a ∈ x⟩⟩ ⟨⟨y, hy⟩, ⟨b, hb : b ∈ y⟩⟩ hab
+    dsimp at hab
+    rw [hab] at ha
+    rw [Sigma.subtype_ext_iff]
+    simp only [Subtype.mk_eq_mk, Subtype.coe_mk]
+    apply And.intro _ hab
+    exact ExistsUnique.unique (hc.2 b) ⟨hx, ha⟩ ⟨hy, hb⟩
+  · -- surjectivity
+    intro a
+    obtain ⟨x, ⟨hx, ha : a ∈ x⟩, _⟩ := hc.2 a
+    use ⟨⟨x, hx⟩, ⟨a, ha⟩⟩
```

Comment:
Is this lemma used anywhere in the PR? If not, you could move it to a separate PR.

Reply:
In fact, I don't use it anymore. Thanks for noting.